### PR TITLE
refactor(prf): mpz upgrade

### PR DESCRIPTION
**Pull request by sinui0**
This PR updates our `prf` crate to use the new `mpz` API.

*Note*: The exact commit of `mpz` is not settled yet, we will update this prior to release.

# Changes
- A bit of general clean up including making the error type opaque
- Utilizes the preprocessing API from `mpz` to preprocess the prf as much as possible prior to the online phase.
- The PRF verify data was switched to being public, which should cut down on latency due to not needing OTPs.

### CODE REVIEW

File: components/prf/hmac-sha256/src/prf.rs [hunk 1]
```diff
@@ -157,36 +196,38 @@ where
...
-            std::mem::replace(&mut self.state, state::State::Error).try_into_server_finished()?;
+        let State::ServerFinished { hash_state, sf_vd } = self.state.take() else {
+            return Err(PrfError::state("PRF not in server finished state"));
+        };
 
         let circ = SERVER_VD_CIRC.get().expect("server vd circuit is set");
 
         if let Some(handshake_hash) = handshake_hash {
-            self.thread_1
+            self.thread_0
```
Comment (th4s) [inline components/prf/hmac-sha256/src/prf.rs:226]: This is now `thread_0` instead of `thread_1` but it doesn't matter, right?

**Review by th4s — COMMENTED** [review]
Nice :+1:

Shouldn't we adapt the `Prf` trait to reflect that `cfvd` and `sfvd` are no longer private?

File: components/prf/hmac-sha256/src/prf.rs [hunk 2]
```diff
@@ -199,17 +240,17 @@ where
             .await?;
 
         let vd = if handshake_hash.is_some() {
-            let mut outputs = self.thread_1.decode_private(&[sf_vd.vd]).await?;
+            let mut outputs = self.thread_0.decode(&[sf_vd.vd]).await?;
```
Comment (themighty1) [inline components/prf/hmac-sha256/src/prf.rs:243]: as above, we need to make sure that `compute_server_finished_vd_private/blind` is now called `compute_server_finished_vd`.

**Review by themighty1 — CHANGES_REQUESTED** [review]
gw! some changes may be needed.

**Review by themighty1 — APPROVED** [review]

**Pull request merged**